appstore-metadata-service/src/test/java/com/lgi/appstore/metadata/api/delegate/TestContainerPostgresSQLDelegate.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2023 Liberty Global Technology Services BV
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.lgi.appstore.metadata.api.delegate;
+
+import org.testcontainers.containers.ContainerState;
+import org.testcontainers.delegate.AbstractDatabaseDelegate;
+import org.testcontainers.exception.ConnectionCreationException;
+import org.testcontainers.ext.ScriptUtils;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.util.Properties;
+
+public class TestContainerPostgresSQLDelegate extends AbstractDatabaseDelegate<Connection> {
+    private final static String USER = "user";
+    private final static String PASSWORD = "password";
+    private final static String JDBC_URL_TEMPLATE = "jdbc:postgresql://localhost:%s/test";
+    private final static String CONNECTION_CREATION_EXCEPTION_MSG = "Could not obtain PostgresSQL connection";
+    private final ContainerState container;
+    private String user;
+    private String password;
+    public TestContainerPostgresSQLDelegate(ContainerState container, String user, String password) {
+        this.container = container;
+        this.user = user;
+        this.password = password;
+    }
+
+    @Override
+    protected Connection createNewConnection() {
+        try {
+            Properties connectionProps = new Properties();
+            connectionProps.put(USER, user);
+            connectionProps.put(PASSWORD, password);
+            return DriverManager.getConnection(
+                    String.format(JDBC_URL_TEMPLATE, container.getFirstMappedPort()),
+                    connectionProps);
+        } catch (Exception e) {
+            throw new ConnectionCreationException(CONNECTION_CREATION_EXCEPTION_MSG, e);
+        }
+    }
+
+    @Override
+    public void execute(String statement, String scriptPath, int lineNumber, boolean continueOnError, boolean ignoreFailedDrops) {
+        try {
+            ResultSet result = getConnection().prepareStatement(statement).executeQuery();
+            result.next();
+            if (!result.getObject(1, Integer.class).equals(1)) {
```

Comment:
I think we should move that logic to `PostgresContainerInitializer`.  That method accepts any SQL query and non-necessary it will return `1`.

Reply:
Created separated `selectOne` method and there we don't need to check the result from the execution.